### ISSUE TITLE
tf$device("/cpu0") or tf$device("/gpu0") result in "IndexError: list index out of range"

### ISSUE THREAD

**Issue by antspengy** [rstudio/tensorflow#272]
Firstly, great package and documentation so thanks a lot for putting this all together.

Keras and tensorflow in R have been working well for me in cpu and gpu mode on both  local Windows 10 and remote Azure Linux Ubuntu 16.04 installations.  I'm now at the point of experimenting with multiple gpu model building and have been following instructions from [RStudio Tensorflow multi_gpu model reference](https://tensorflow.rstudio.com/keras/reference/multi_gpu_model.html).  However I keep encountering an _IndexError_ error with all references to tf$device in my code.  For instance, the code: 
`library(tensorflow)`
`tf_config()`
`sess = tf$Session()`
`hello <- tf$constant('Hello, TensorFlow!')`
`sess$run(hello)`

`with(tf$device("/cpu:0"), {`
`  const <- tf$constant(42)})`

results in the following console output:

> [1] "Hello, TensorFlow!"
> TensorFlow v1.11.0  (~/.virtualenvs/r-reticulate/local/lib/python2.7/site-packages/keras)  

> Python  v2.7 (~/.virtualenvs/r-reticulate/bin/python)

> Error in py_call_impl(callable, dots$args, dots$keywords) : 
>   IndexError: list index out of range
> Detailed traceback: 
>   File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
>     return self.gen.next()
>   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 4243, in device
>     self._add_device_to_stack(device_name_or_function, offset=2)
>   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 4182, in _add_device_to_stack
>     self._device_function_stack.push_obj(spec, offset=total_offset)
>   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/traceable_stack.py", line 106, in push_obj
>     return traceable_obj.set_filename_and_line_from_caller(offset + 1)
>   File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/traceable_stack.py", line 64, in set_filename_and_line_from_caller
>     self.filename, self.lineno = frame_records[negative_offset][:2]

However in the same session I can run equivalent code in python without a problem:
`library(reticulate)`
`r_constant <- py_run_string("`
`import tensorflow as tf`
`with tf.device('/cpu:0'):`
`        const = tf.constant(42)")`
`print(r_constant)`

and it appears to be perfectly fine with output:
> {'const': <tf.Tensor 'Const_5:0' shape=() dtype=int32>, '__builtins__': <module '__builtin__' (built-in)>, '__package__': None, 'sys': <module 'sys' (built-in)>, 'R': <class '__main__.R'>, 'tf': <module 'tensorflow' from '/usr/local/lib/python2.7/dist-packages/tensorflow/__init__.pyc'>, '__name__': '__main__', 'r': <__main__.R>, '__doc__': None}

Note that all other aspects of the tensorflow and keras applications seem to be working well, and I haven't been able to resolve this specific issue even when using tensorflow 1.09 or 1.10 or using install_tensorflow with my anaconda or virtual instances using Python 3.5 and 3.6.  The same issue occurs when I use 'gpu:0' as well.  I haven't encountered any problems doing this operational natively in python in jupyter notebook instances on the same environments either. 

Any thoughts/suggestions would be much appreciated, thanks.

**Comment by skeydan** [comment]
thanks! I can in fact reproduce when I use an env that has a GPU.

To be followed up in

https://github.com/rstudio/reticulate/issues/365

**Comment by jjallaire** [comment]
Let's wait to remove the 1.10 by default until they actually ship this in
v1.12.

J.J.

**Comment by skeydan** [comment]
Hi Javier, thanks for taking care of this!

I just test on a fresh (as of last night) build of 1.12 though and I still get the error:

```
> with(tf$device("/cpu:0"), {
+   const <- tf$constant(42)
+ })
Error in py_call_impl(callable, dots$args, dots$keywords) : 
  IndexError: list index out of range

Detailed traceback: 
  File "/home/key/anaconda3/envs/tf-12-gpu/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/key/anaconda3/envs/tf-12-gpu/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 4245, in device
    self._add_device_to_stack(device_name_or_function, offset=2)
  File "/home/key/anaconda3/envs/tf-12-gpu/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 4184, in _add_device_to_stack
    self._device_function_stack.push_obj(spec, offset=total_offset)
  File "/home/key/anaconda3/envs/tf-12-gpu/lib/python3.6/site-packages/tensorflow/python/framework/traceable_stack.py", line 106, in push_obj
    return traceable_obj.set_filename_and_line_from_caller(offset + 1)
  File "/home/key/anaconda3/envs/tf-12-gpu/lib/python3.6/site-packages/tensorflow/python/framework/traceable_stack.py", line 64, in set_filename_and_line_from_caller
    self.filename, self.lin
```

Looks like they merged it into master but not branch 1.12...

**Comment by jjallaire** [comment]
I think we should stay with 1.10 until 1.13 comes out.
